### PR TITLE
zynq7000: add functions to set clocks and pins configuration

### PR DESCRIPTION
**Pull request by HBuczynski**

### CODE REVIEW

File: armv7a9-zynq7000/zynq.c [hunk 1]
```diff
@@ -119,6 +116,271 @@ struct {
...
+
+		case ctrl_can_clk:
+			*(zynq_common.slcr + slcr_can_clk_ctrl) = (*(zynq_common.slcr + slcr_can_clk_ctrl) & ~0x03f03f33) | (!!clk->pll.clkact0) |
+                                                      ((!!clk->pll.clkact1) << 1) | ((clk->pll.srcsel & 0x3) << 4) | ((clk->pll.divisor0 & 0x3f) << 8) |
+                                                      ((clk->pll.divisor1 & 0x3f) << 20);
+			break;
+
+		case ctrl_can_mioclk:
+			*(zynq_common.slcr + slcr_can_mioclk_ctrl) = (*(zynq_common.slcr + slcr_can_mioclk_ctrl) & ~0x007f007f) | (clk->mio.mux0 & 0x3f) | ((!!clk->mio.ref0) << 6) |
+                                                         ((clk->mio.mux1 & 0x3f) << 16) | ((!!clk->mio.ref1) << 22);
```
Comment (agkaminski) [inline armv7a9-zynq7000/zynq.c:219]: Single additional indent is enough when breaking a line

File: armv7a9-zynq7000/zynq.c [hunk 2]
```diff
@@ -119,6 +116,271 @@ struct {
...
+	return 0;
+}
+
+
+int _zynq_getCtlClock(ctl_clock_t *clk)
+{
+	u32 id;
+	u32 val = 0;
+
+	_zynq_slcrUnlock();
```
Comment (agkaminski) [inline armv7a9-zynq7000/zynq.c:238]: Do we need to unlock when reading a value?

**Review by agkaminski — COMMENTED** [review]
Apart from this one comment LGTM

File: armv7a9-zynq7000/zynq.c (hunk 2 above)
Comment (HBuczynski) [reply armv7a9-zynq7000/zynq.c]: Good note, I've tested it and it works without unlocking slcr.

**Pull request merged**